proposal/calculation-api.d.ts

```diff
@@ -0,0 +1,193 @@
+   *   error.
+   *
+   * - Let `calculation` be a calculation with name `"min"` and `arguments` as
+   *   its arguments.
+   *
+   * - Return the result of [simplifying] `calculation`.
+   *
+   *   [simplifying]: ../spec/types/calculation.md#simplifying-a-calculation
+   */
+  static min(
+    arguments: CalculationValue[] | List<CalculationValue>
+  ): SassNumber | SassCalculation;
+
+  /**
+   * Creates a value that represents `min(...arguments)`.
+   *
+   * - If `arguments` transitively contains a quoted `SassString`, throw an
+   *   error.
+   *
+   * - Let `calculation` be a calculation with name `"max"` and `arguments` as
+   *   its arguments.
+   *
+   * - Return the result of [simplifying] `calculation`.
+   *
+   *   [simplifying]: ../spec/types/calculation.md#simplifying-a-calculation
+   */
+  static max(
+    arguments: CalculationValue[] | List<CalculationValue>
+  ): SassNumber | SassCalculation;
+
+  /**
+   * Creates a value that represents `calc(min, value, max)` expression.
+   *
+   * - If `min`, `max`, or `clamp` is or transitively contains a quoted
+   *   `SassString`, throw an error.
+   *
+   * - If `value` is undefined and `max` is not undefined, throw an error.
+   *
+   * - If `value` or `max` is undefined and neither `min` nor `value` is a
+   *   `SassString` that begins with `"var("`, throw an error.
+   *
+   * - Let `calculation` be a calculation with name `"clamp"` and `min`,
+   *   `value`, and `max` as its arguments, excluding any arguments that are
+   *   undefined.
+   *
+   * - Return the result of [simplifying] `calculation`.
+   *
+   *   [simplifying]: ../spec/types/calculation.md#simplifying-a-calculation
+   */
+  static clamp(
+    min: CalculationValue,
+    value?: CalculationValue,
+    max?: CalculationValue
+  ): SassNumber | SassCalculation;
+
+  /** `internal`'s `name` field. */
+  get name(): string;
+
+  /** A list of `internal`'s arguments. */
+  get arguments(): List<CalculationValue>;
+}
+
+/** The set of possible operators in a Sass calculation. */
+type CalculationOperator = '+' | '-' | '*' | '/';
+
+/**
+ * The JS API representation of a Sass [CalculationOperation].
+ *
+ * [CalculationOperation]: ../spec/types/calculation.md#types
+ *
+ * `internal` refers to a Sass CalculationOperation.
+ */
+export abstract class CalculationOperation implements ValueObject {
+  /**
+   * Creates a Sass CalculationOperation by setting the fields to the arguments
+   * of the corresponding names, and returns it.
+   */
+  constructor(
+    operator: CalculationOperator,
+    left: CalculationValue,
+    right: CalculationValue
+  );
+
+  /** `internal`'s `operator` field. */
+  get operator(): CalculationOperator;
+
+  /** `internal`'s `left` field. */
+  get left(): CalculationValue;
+
+  /** `internal`'s `right` field. */
+  get right(): CalculationValue;
+
+  /** Whether `internal` is equal to `other.internal` in Sass. */
```

Comment:
the description looks weird to me, as there is no `internal` in this type definition.

Reply:
See https://github.com/sass/sass/blob/32a7db88ee8d320ee92e6e0deb4f85e21696ba77/spec/js-api/value/index.d.ts#L32-L34